src/detect-engine-prefilter.h

```diff
@@ -33,6 +33,8 @@ typedef struct DetectTransaction_ {
     const uint64_t tx_id;
     struct AppLayerTxData *tx_data_ptr;
     DetectEngineStateDirection *de_state;
+    // state for bidirectional signatures
+    DetectEngineStateDirection *de_state_bidir;
```

Comment:
I see `de_state` also being used for bidir?

Reply:
A short-lived `DetectTransaction` has both a state for unidirectional transactions and bidirectional txs that is a `DetectEngineStateDirection` 

We call `RuleMatchCandidateMergeStoredState` on both

It is not the same thing as the `DetectEngineState *de_state;` in `AppLayerTxData` which is long-lived 

Did you have confusion with both these same-named fields but representing different structures ?

Reply:
Good question...

Looks like your proposal passes my SV tests...

Will use it